modules/basics/src/main/java/com/opengamma/strata/basics/market/ReferenceData.java

```diff
@@ -89,6 +89,31 @@ public default boolean containsValue(ReferenceDataId<?> id) {
   }
 
   /**
+   * Gets the reference data value associated with the specified identifier, cast to a specific type.
+   * <p>
+   * If this reference data instance contains the identifier, and the value is of the correct type,
+   * then the value will be returned. Otherwise, an exception will be thrown.
+   * 
+   * @param <T>  the type of the reference data
+   * @param <S>  the type of the reference data subclass
+   * @param id  the identifier to find
+   * @param type  the type of the subclass
+   * @return the reference data value
+   * @throws ReferenceDataNotFoundException if the identifier is not found
+   * @throws ClassCastException if the identifier exists, but is not of the correct type
+   */
+  @SuppressWarnings("unchecked")
+  public default <T, S extends T> S getValue(ReferenceDataId<T> id, Class<S> type) {
```

Comment:
Why is this method necessary? The ID already narrows the type of the returned value. Is it a common requirement to narrow it further? So common that it's worth having on the API rather than the caller downcasting the returned value?